src/lib/utils/mergeSearchParameters.ts

```diff
@@ -103,6 +105,23 @@ const mergeHierarchicalFacetsRefinements: Merger = (left, right) =>
     },
   });
 
+const mergeRuleContexts: Merger = (left, right) => {
+  const ruleContexts: string[] = uniq(
+    ([] as any)
+      .concat(left.ruleContexts)
+      .concat(right.ruleContexts)
+      .filter(Boolean)
+  );
+
+  if (ruleContexts.length > 0) {
```

Comment:
Note that I add the `ruleContexts` key in the search parameters only when not empty. Do we want to always add it?

Reply:
I would say omit when it doesn't make sense.